lom/src/engine/context.go

```diff
@@ -0,0 +1,394 @@
+package engine
+
+import (
+    . "lom/src/lib/lomcommon"
+    . "lom/src/lib/lomipc"
+    "sort"
+    "time"
+)
+/*
+ * Maintains current engine context as all active clients, their registrations,
+ * outstanding requests for client to read/write.
+ *
+ * All clients requests from clients are processed here.
+ *
+ * Manage incoming heartbeats
+ */
+
+/* Action info is collected from register calls from clients */
+type ActiveActionInfo_t struct {
+    Action      string
```

Comment:
General Question. If proc.conf is incorrect like having two same action names part of different procs. Is it handly currently in engine ?

Reply:
Engine does not handle procs.conf
More over, you can write lib code such that the later entry overwrites. So in case of duplicates the last instance wins.